src/Serializer/AbstractItemNormalizer.php

```diff
@@ -122,14 +134,33 @@ protected function extractAttributes($object, $format = null, array $context = [
     /**
      * {@inheritdoc}
      */
-    protected function getAllowedAttributes($classOrObject, array $context, $attributesAsString = false)
-    {
+    protected function getAllowedAttributes(
+        $classOrObject,
+        array $context,
+        $attributesAsString = false
+    ) {
         $options = $this->getFactoryOptions($context);
-        $propertyNames = $this->propertyNameCollectionFactory->create($context['resource_class'], $options);
+        $propertyNames = $this
+            ->propertyNameCollectionFactory
+            ->create($context['resource_class'], $options);
 
         $allowedAttributes = [];
         foreach ($propertyNames as $propertyName) {
-            $propertyMetadata = $this->propertyMetadataFactory->create($context['resource_class'], $propertyName, $options);
+            $propertyMetadata = $this
+                ->propertyMetadataFactory
+                ->create($context['resource_class'], $propertyName, $options);
+
+            if (
+                isset($context['api_denormalize'])
+                    && !$propertyMetadata->isWritable()
+                    && !$propertyMetadata->isIdentifier()
+            ) {
+                throw new RuntimeException(sprintf(
+                    'Property \'%s.%s\' is not writeable',
+                    $context['resource_class'],
+                    $propertyName
+                ));
+            }
```

Comment:
I made this because I spent quite some time figuring out why the `age` field in `ParentDummy` was not being updated when I patched an entity in my tests.

Turned out it was because the field was private and didn't have a setter, so it not 'writeable'. It would have helped to know that instead of simply ignoring the field, so I thought it'd be nice to get an error if trying to update (or post) to a non-writeable file.

I'll revert it. If you think is a valid point I'll open another PR :-)